src/main/java/ch/njol/skript/expressions/ExprStringColor.java

```diff
@@ -0,0 +1,151 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.SkriptColor;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Name("String Colors")
+@Description({
+	"Retrieve the first, the last, or all of the colors of a string.",
+	"NOTE: The retrieved colors of the string will be formatted with the color symbol."
+})
+@Examples({
+	"set {_colors::*} to the string colors of \"<red>hey<blue>yo\"",
+	"",
+	"set {_color} to the first string color of \"&aGoodbye!\"",
+	"send \"%{_color}%Howdy!\" to all players"
+})
+@Since("INSERT VERSION")
+public class ExprStringColor extends PropertyExpression<String, String> {
+
+	private enum StringColor {
+		ALL("[all [of]] [the]", "all the"),
+		FIRST("[the] first", "the first"),
+		LAST("[the] last", "the last");
+
+		private String pattern, toString;
+
+		StringColor(String pattern, String toString) {
+			this.pattern = pattern;
+			this.toString = toString;
+		}
+	}
+
+	private static final StringColor[] STRING_COLORS = StringColor.values();
+
+	static {
+		String[] patterns = new String[STRING_COLORS.length];
+		for (StringColor color : STRING_COLORS) {
+			patterns[color.ordinal()] = color.pattern + " string color[s] of %strings%";
+		}
```

Comment:
can we not have the patterns hidden away in the enum?
It's much easier to spot issues/know syntax at a glance if it's written out in the register method.
There's very little benefit to putting it in the enum here, imo